utils.go

```diff
@@ -42,43 +41,39 @@ const (
 	maxBPFClassifierNameLen = 256
 )
 
-// availableFilterFunctions - cache of the list of available kernel functions.
-var availableFilterFunctions []string
-
 func FindFilterFunction(funcName string) (string, error) {
 	// Prepare matching pattern
 	searchedName, err := regexp.Compile(funcName)
 	if err != nil {
 		return "", err
 	}
 
-	// Cache available filter functions if necessary
-	if len(availableFilterFunctions) == 0 {
-		funcs, err := tracefs.ReadFile("available_filter_functions")
-		if err != nil {
-			return "", err
-		}
-		availableFilterFunctions = strings.Split(string(funcs), "\n")
-		for i, name := range availableFilterFunctions {
-			splittedName := strings.Split(name, " ")
-			name = splittedName[0]
-			splittedName = strings.Split(name, "\t")
-			name = splittedName[0]
-			availableFilterFunctions[i] = name
-		}
-		sort.Strings(availableFilterFunctions)
+	funcsReader, err := tracefs.Open("available_filter_functions")
+	if err != nil {
+		return "", err
 	}
+	defer funcsReader.Close()
+
+	funcs := bufio.NewScanner(funcsReader)
+	funcs.Split(bufio.ScanLines)
 
-	// Match function name
 	var potentialMatches []string
-	for _, f := range availableFilterFunctions {
-		if searchedName.MatchString(f) {
-			potentialMatches = append(potentialMatches, f)
+	for funcs.Scan() {
+		name := funcs.Text()
+		name, _, _ = strings.Cut(name, " ")
+		name, _, _ = strings.Cut(name, "\t")
+
+		if searchedName.MatchString(name) {
+			potentialMatches = append(potentialMatches, name)
 		}
-		if f == funcName {
-			return f, nil
+		if name == funcName {
+			return name, nil
```

Comment:
```suggestion
		name := funcs.Bytes()
		name, _, _ = bytes.Cut(name, []byte(" "))
		name, _, _ = bytes.Cut(name, []byte("\t"))

		if string(name) == funcName {
			return funcName, nil
		}
		if searchedName.Match(name) {
				potentialMatches = append(potentialMatches, string(name))
		}
```

Reply:
I think this should cut the allocations down ever more.

Reply:
The diff is wonky for some reason, so don't apply directly from Github.

Reply:
It's much better especially from the alloc point of view, thanks !